### PR TITLE
39 add the game engine functionality

### LINKED ISSUE #39: Add the game engine functionality

### PR DESCRIPTION
**Pull request by mmmtastymmm**

### CODE REVIEW

**Comment by mmmtastymmm** [comment]
Coverage is hard again but this sets up the web socket logic

File: src/table/mod.rs [hunk 1]
```diff
@@ -354,7 +358,7 @@ impl Table {
         result_string
     }
 
-    /// Cleans all the cards from the table
+    /// Cleans all the cards from_server_url the table
```
Comment (corbincallahan) [inline src/table/mod.rs]: Reckless replace all smh

Comment (mmmtastymmm) [reply src/table/mod.rs]: Oopsie

Comment (mmmtastymmm) [reply src/table/mod.rs]: I blame rust rover for giving me too much power

**Pull request merged**